addons/mrp/migrations/11.0.2.0/post-migration.py

```diff
@@ -0,0 +1,207 @@
+            picking_id,
+            product_id,
+            product_qty,
+            product_uom_id,
+            product_uom_qty,
+            qty_done,
+            reference,
+            state,
+            write_date,
+            write_uid,
+            done_move,
+            done_wo,
+            lot_produced_id,
+            lot_produced_qty,
+            production_id,
+            workorder_id"""
+    select = """current_timestamp,
+            sm.write_uid,
+            sm.date::date,
+            sm.location_dest_id,
+            sm.location_id,
+            sml.lot_id,
+            spl.name,
+            sm.id,
+            sm.%s,
+            sm.restrict_partner_id,
+            sm.picking_id,
+            sm.product_id,
+            CASE WHEN sm.is_done = TRUE THEN 0
+            ELSE sm.product_qty
+            END as product_qty,
+            sm.product_uom,
+            CASE WHEN sm.is_done = TRUE THEN 0
+            ELSE sm.product_uom_qty
+            END as product_uom_qty,
+            sm.product_uom_qty as qty_done,
+            sm.name,
+            sm.state,
+            current_timestamp,
+            sm.write_uid,
+            sm.is_done,
+            COALESCE(sml.done_wo, TRUE),
+            sml.lot_produced_id,
+            sml.lot_produced_qty,
+            sm.raw_material_production_id,
+            sml.workorder_id""" % openupgrade.get_legacy_name(
+        'quantity_done_store')
+    from_ = """stock_move sm
+        INNER JOIN mrp_production mp ON sm.raw_material_production_id = mp.id
+        LEFT JOIN stock_move_lots sml ON (sml.move_id = sm.id
+            AND sml.production_id = mp.id)
+        LEFT JOIN stock_production_lot spl ON sml.lot_id = spl.id"""
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO stock_move_line (%(insert_into)s
+        )
+        SELECT %(select)s
+        FROM %(from)s
+        WHERE sm.state NOT IN ('cancel', 'confirmed')
+            AND sm.id NOT IN (SELECT sq.reservation_id
+                              FROM stock_quant sq
+                              WHERE sq.reservation_id IS NOT NULL)
+        """ % {
+            'insert_into': insert_into,
+            'select': select,
+            'from': from_,
+        },
+    )
+    select = """current_timestamp,
+            sm.write_uid,
+            sm.date::date,
+            sm.location_dest_id,
+            sm.location_id,
+            sml.lot_id,
+            spl.name,
+            sm.id,
+            0,
+            sm.restrict_partner_id,
+            sm.picking_id,
+            sm.product_id,
+            0,
+            sm.product_uom,
+            0,
+            sm.product_uom_qty as qty_done,
+            sm.name,
+            sm.state,
+            current_timestamp,
+            sm.write_uid,
+            sm.is_done,
+            TRUE,
+            sml.lot_produced_id,
+            sml.lot_produced_qty,
+            NULL,
+            sml.workorder_id"""
+    from_ = """stock_move sm
```

Comment:
Have you checked if there's some kind of sml duplication from standard `stock` inserts?

Reply:
It's impossible. There is an inner join in the following line that ensures that.

Reply:
I didn't found any duplication :S

Reply:
OK, I trust you on this :smiley: